applications/forms.py

```diff
@@ -176,7 +255,56 @@ def fieldsets(self):
+                                  'description': 'Now tell us some information we need for your volunteer application'
+                                  }),
+                                ('Volunteer Info',
+                                 {'fields': ('first_timer', 'first_timer_extra', 'which_hack', 'attendance',
+                                             'english_level', 'quality', 'weakness', 'cool_skill', 'fav_movie',
+                                             'lennyface', 'friends'),
+                                  'description': "Now we're going to ask you some questions about your volunteering "
+                                                 "skills."
+                                  })
+                                ))
+
+    def __sponsorFieldsets(self):
+        self._fieldsets.append((('Sponsor Info',
+                                 {'fields': ('company', 'sponsor_position', 'attendance'),
+                                  'description': 'Tell us some information we need for your volunteer application'})))
+
+    def __mentorFieldsets(self):
+        self._fieldsets.extend((('Mentor Info', {'fields': ('description', 'origin', 'is_student', 'company',
+                                                            'university', 'degree', 'first_timer', 'first_timer_extra',
+                                                            'which_hack', 'quality', 'english_level', 'attendance',
+                                                            'lennyface'),
+                                                 'description': 'Tell us some information we need for your mentor '
+                                                                'application'}),
+                                ('Show us what you\'ve built',
+                                 {'fields': ('github', 'devpost', 'linkedin', 'site', 'resume'),
+                                  'description': 'Some of our sponsors may use this information for recruitment '
+                                                 'purposes, '
+                                                 'so please include as much as you can.'})
+                                ))
+
+    def fieldsets(self):
+        # Fieldsets ordered and with description
+        self._fieldsets = [('Personal Info',
+                            {'fields': ('gender', 'other_gender', 'under_age', 'phone_number', 'tshirt_size', 'diet',
+                                        'other_diet'),
+                             'description': 'Hey there, before we begin we would like to know a little more about '
+                                            'you.'})]
+        if self.instance.is_hacker:
+            self.__hackerFieldsets()
+        elif self.instance.is_volunteer:
+            self.__volunteerFieldsets()
+        elif self.instance.is_mentor:
+            self.__mentorFieldsets()
+        elif self.instance.is_sponsor:
+            self.__sponsorFieldsets()
```

Comment:
This is hacky. Pls separate this into separate models for each type of application.